core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -52,13 +56,14 @@ public abstract class ManifestWriter<F extends ContentFile<F>> implements FileAp
   private long deletedRows = 0L;
   private Long minDataSequenceNumber = null;
 
-  private ManifestWriter(PartitionSpec spec, OutputFile file, Long snapshotId) {
-    this.file = file;
+  private ManifestWriter(PartitionSpec spec, EncryptedOutputFile file, Long snapshotId) {
+    this.file = file.encryptingOutputFile();
     this.specId = spec.specId();
-    this.writer = newAppender(spec, file);
+    this.writer = newAppender(spec, file.encryptingOutputFile());
     this.snapshotId = snapshotId;
     this.reused = new GenericManifestEntry<>(spec.partitionType());
     this.stats = new PartitionSummary(spec);
+    this.keyMetadataBuffer = file.keyMetadata().buffer();
```

Comment:
Are you sure that this is always empty? I think it could be null and calling `buffer` would raise a `NullPointerException`.

Reply:
In the writer, we pass the `EncryptionKeyMetadata.EMPTY` object for unencrypted manifests. But to be on the safe side, we can also check for null objects.